### PR TITLE
MetricsTestKit: expose all metrics publicly. This allows users to run checks on counters when they might not know the exact label

### PR DESCRIPTION
**Pull request by hamzahrmalik**
Provide functions on TestMetrics to retrieve all counters/timers/recorders

### Motivation:

Users may want to iterate over metrics to perform some complex assertions. For example, they might want to confirm that no metrics have been emitted with a certain dimension

### Modifications:

Added functions for allRecorders, allTimers and allCounters which return arrays of TestRecorder, TestTimer and TestCounter

### CODE REVIEW

**Comment by ktoso** [comment]
@swift-server-bot add to allowlist

**Review by ktoso — APPROVED** [review]
In general I think this is fine, as a testing facility.

Thanks for the PR!

**Review by tomerd — APPROVED** [review]

**Review by tomerd — APPROVED** [review]
are there any tests we need to add?

File: Sources/MetricsTestKit/TestMetrics.swift [hunk 1]
```diff
@@ -155,15 +155,21 @@ extension TestMetrics {
...
         }
-        guard let testCounter = maybeCounter as? TestCounter else {
-            throw TestMetricsError.illegalMetricType(metric: maybeCounter, expected: "\(TestCounter.self)")
-        }
         return testCounter
     }
 
+    /// Use this function to run assertions on which counters have been created
+    /// - Returns: All the counters which have been created and not destroyed
+    func allCounters() throws -> [TestCounter] {
```
Comment (FranzBusch) [inline Sources/MetricsTestKit/TestMetrics.swift:166]: This is not `public`

**Review by FranzBusch — CHANGES_REQUESTED** [review]
Shouldn't we have a `allGauges` and now also an `allMeters` method?

**Comment by hamzahrmalik** [comment]
> This has a merge conflict now due to the introduced meters on main, could you rebase @hamzahrmalik ?
> 
> Otherwise LGTM so let's merge this and the follow up which adds tests to this 👍

I actually think we should merge #128 first
That one is simply a refactor, and will make this one very simple (just adding public modifiers)

**Comment by ktoso** [comment]
Yeah that works, merged #128 👍

**Pull request merged**